File: src/types/prerequisites/single/SexualCharacteristicPrerequisite.ts
import * as DB from "tsondb/schema/dsl"

export const SexualCharacteristicPrerequisite = DB.TypeAlias(import.meta.url, {
  name: "SexualCharacteristicPrerequisite",
  comment: "Requires a specific sexual characteristic.",
  type: () =>
    DB.Object({
      id: DB.Required({
        type: DB.IncludeIdentifier(SexualCharacteristic),
      }),
    }),
})

const SexualCharacteristic = DB.Enum(import.meta.url, {
  name: "SexualCharacteristic",
  values: () => ({
    Penis: DB.EnumCase({ type: null }),
    Vagina: DB.EnumCase({ type: null }),
  }),
})

File: Sources/OptolithDatabaseSchema/Types/prerequisites/single/SexualCharacteristicPrerequisite.swift
import FileDB

/// Requires a specific sexual characteristic.
@Embedded
public struct SexualCharacteristicPrerequisite {
      id: Required({
        type: IncludeIdentifier(SexualCharacteristic),
      }),
  }

@ModelEnum
public enum SexualCharacteristic {
    case Penis
    case Vagina
}
